src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -485,6 +509,23 @@ public PartitionIterator fetchPage(PageSize pageSize, long queryStartNanoTime)
                 return pager.fetchPageInternal(pageSize, executionController);
             }
         }
+
+        public static class OffsetPager extends Pager
+        {
+            private final Pager pager;
+
+            private OffsetPager(Pager pager)
+            {
+                super(pager.pager);
+                this.pager = pager;
+            }
+
+            @Override
+            public PartitionIterator fetchPage(PageSize pageSize, long queryStartNanoTime)
+            {
+                return new PagerExhaustingIterator(pager, pageSize, queryStartNanoTime);
```

Comment:
Can it be called multiple times?

Reply:
I think it can be called multiple times, it simply will return an empty iterator because the underlying `QueryPager` is already exhausted. 
Independently of offsets, `SelectStatement.Pager` is tricky because one might think that it's meant to be used with multiple calls to `fetchPage` inside a loop that checks `Pager#isExhausted`. However, we only have a single call to `fetchPage` for any kind of pager instance. That single call fetches the single page of results that will be sent to client for that `SELECT` query execution. The next pages will be requested by further `SELECT` statements run by the client, using a new `Pager` instance per page.